### PR TITLE
Modify Get-DbaService to allow collecting Clustered state and Version information

### PR DESCRIPTION
**Pull request by sqllensman**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [X] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [X] Pester test is included
 - [ ] Nunit test is included
 - [X] Documentation
 - [ ] Build system
 

### Purpose
Adds an option to request additional information about SQL Server related services.
By default the command only gets data from the SqlService namespace. Additional information is available from the SqlServiceAdvancedProperty namespace. In particular information can be collected about Clustered status of Services, the software version and Sku.

### Approach
Adds a new switch to allow collecting data from SqlServiceAdvancedProperty namespace

### Commands to test
Get-DbaService

### Learning
Based on ideas in
https://www.mssqltips.com/sqlservertip/2358/using-powershell-to-pull-sql-server-services-configuration-information/

and 

https://github.com/proxb/PowerShell_Scripts/blob/master/Get-PSSQLInstance.ps1

### CODE REVIEW

**Comment by potatoqualitee** [comment]
fabulous! thanks. I'll ask @nvarscar to take a look and approve for merging.

File: functions/Get-DbaService.ps1 [hunk 1]
```diff
@@ -219,8 +230,40 @@
                                     Set-ServiceStartMode -InputObject $this -Mode $Mode -ErrorAction Stop
                                     $this.StartMode = $Mode
                                 }
-                                Select-DefaultView -InputObject $service -Property ComputerName, ServiceName, ServiceType, InstanceName, DisplayName, StartName, State, StartMode -TypeName DbaSqlService
+
```
[the remark this answers is not in the document]
Comment (sqllensman) [reply functions/Get-DbaService.ps1]: Moved the code into this block

File: functions/Get-DbaService.ps1 [hunk 2]
```diff
@@ -219,8 +230,40 @@
...
+                            }
+
+                            if ($AdvancedProperties) {
+                                Add-Member -Force -InputObject $service -MemberType NoteProperty -Name AdvancedRequested -Value 1
+                                if($namespaces.count -eq 1) {
+                                    $namespaceValue = $namespaces[0].Name
+                                }
+                                else {
+                                    $namespaceValue = $service.CimClass.ToString().ToUpper().Replace(":SQLSERVICE", "").Replace("ROOT/MICROSOFT/SQLSERVER/","")
```
[the remark this answers is not in the document]
Comment (sqllensman) [reply functions/Get-DbaService.ps1]: Made suggested change

File: functions/Get-DbaService.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (sqllensman) [reply functions/Get-DbaService.ps1]: Code has been moved into the InstanceName block as per suggestion

**Comment by sqllensman** [comment]
I tried to get rid of the *Copy-DbaDbTableData.Tests.ps1* by replacing with current code from Development branch but this does not seem to work.

Not sure how to remove this file.

**Comment by potatoqualitee** [comment]
ill try one moment

**Comment by potatoqualitee** [comment]
Got it ! We had to update your branch with the current version of dev 👍

**Comment by sqllensman** [comment]
Thanks!

**Comment by potatoqualitee** [comment]
fantastic, thanks so much to you both 👑 👑

**Pull request merged**